### ISSUE TITLE
Rework of bullet system.

### ISSUE THREAD

**Issue by andriiqx** [MCModderAnchor/TACZ#2]
In TACZ, every bullet is just the same entity, but with different data (we can't see it)
I'm server owner of ASTRX: NOVA, and it is really important to me to identify every bullet, so i could make datapack/mod for raid system, based on every bullet. But i can't, because, again, every bullet is just the same entity WITHOUT NBT DATA TAGS. 

I'm suggesting to make every bullet as different entity, or just give bullets nbt data like "damage", "name", so developers could identify it in their own tweaks/datapacks/mods. 

On the screenshots you can see, that you can shot RPG rocket, or pistol 9mm bullet, but it would have the same data, and developers wont be able to identify it.
![Screenshot_271](https://github.com/MCModderAnchor/TACZ/assets/116912308/9ec80ee8-448e-4ade-909e-724268fe7c09)
![Screenshot_272](https://github.com/MCModderAnchor/TACZ/assets/116912308/5dba11f1-7af6-4691-802f-09aee3f5e932)

what you can make:
for example, if you shot from AK-47 (it's example, it can be any weapon), you can summon tacz:bullet entity, and gave it next nbt tags:

"ammo": "tacz:762x39",

you can pick "ammo" from data json file of ak-47 (for example)
![Screenshot_273](https://github.com/MCModderAnchor/TACZ/assets/116912308/fe8054e9-d74d-435b-ad2e-f5f681ca2921)
so basically you can just add the name of ammo in nbt tags of bullet entity
so we (developers) will be able to identify it.. now there's not any methods to compare 9mm ammo and rpg7 rocket, because they're all the same entity: tac:bullet. but if you gave nbt tag "name" to every one of them, it would be amazing

**Comment by JackLiang12** [comment]
If you don't mind me asking why you would need to know every bullet?

**Comment by andriiqx** [comment]
how's the progress?

**Comment by TartaricAcid** [comment]
No progress

Bullet entities are not currently stored persistently on the server, so they don't use nbt data
